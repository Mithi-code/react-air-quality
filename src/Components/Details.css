* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
} 

.weather-img{
  display: flex;
  align-items: center;
  width: 50%;
  margin: auto;
  height: 50vh;
}


.weather-card {
  display: flex;
  font-size: large;
  justify-content: center;
}

.weather-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.weather-info p {
  width: 100%;
  border: 2px solid rgb(161, 46, 161);
  padding: 10px;
}