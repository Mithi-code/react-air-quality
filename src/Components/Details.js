import React from 'react';
import './Details.css';
import { useDispatch, useSelector } from 'react-redux';
import { getweathersData } from '../Redux/Detail';
import { useEffect } from 'react';
import img from '../Assest/venom.jpg'

export const Details = () => {
  const data = useSelector((state) => state.weathersReducer.weathers);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getweathersData(location.search.split('=')[1]));
  }, []);
  return (
    <div className="weather-container">
      <img src={img} className="weather-img"/>
      {data && (
        <div className='weather-card'>
          <div className='weather-info'>
            <p><b>Name:-</b>{data.name}</p>
            <p><b>Visibility:-</b>{data.visibility}</p>
            <p><b>Timezone:-</b>{data.timezone}</p>
            <p><b>Name:-</b>{data.name}</p>
            <p><b>Visibility:-</b>{data.visibility}</p>
            <p><b>Timezone:-</b>{data.timezone}</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default Details;
