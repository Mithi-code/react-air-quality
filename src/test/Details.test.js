import { render, asFragment } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import Details from '../Components/Details';

describe('DetailsPage tests', () => {
  test('should render', () => {
    const {asFragment} = render(
      <Provider store={store}>
          <MemoryRouter initialEntries={[{ pathname:'/', state: { flag: 'India' } }]}>
            <Details />
          </MemoryRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});