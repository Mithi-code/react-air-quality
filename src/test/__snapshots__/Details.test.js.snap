// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage tests should render 1`] = `
<DocumentFragment>
  <div
    class="weather-container"
  >
    <h1>
      Weather info of--
    </h1>
    <img
      alt="flag"
      class="weather-img"
      src="India"
    />
    <div
      class="weather-card"
    >
      <div
        class="weather-info"
      >
        <p>
          <b>
            Name:-
          </b>
           
        </p>
        <p>
          <b>
            Wind-Speed:-
          </b>
            KM/H
        </p>
        <p>
          <b>
            Timezone:-
          </b>
           
        </p>
        <p>
          <b>
            Weather-Forecast:-
          </b>
           
        </p>
        <p>
          <b>
            Pressure:-
          </b>
            atm
        </p>
        <p>
          <b>
            humidity:-
          </b>
           %
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
